django/contrib/postgres/search.py

```diff
@@ -0,0 +1,67 @@
+from django.db.models import F, Field
+from django.db.models.expressions import Func, Value
+from django.db.models.functions import Coalesce
+from django.db.models.lookups import Lookup
+
+
+class SearchVectorExact(Lookup):
+    lookup_name = 'exact'
+
+    def process_rhs(self, qn, connection):
+        if not hasattr(self.rhs, 'output_field') or isinstance(self.rhs.output_field, SearchVectorField):
+            self.rhs = SearchQuery(self.rhs, dictionary=self.lhs.dictionary)
+        rhs, rhs_params = super(SearchVectorExact, self).process_rhs(qn, connection)
+        return rhs, rhs_params
+
+    def as_sql(self, qn, connection):
+        lhs, lhs_params = self.process_lhs(qn, connection)
+        rhs, rhs_params = self.process_rhs(qn, connection)
+        params = lhs_params + rhs_params
+        return '%s @@ %s = true' % (lhs, rhs), params
+
+
+class SearchVectorField(Field):
+
+    def db_type(self, connection):
+        return 'tsvector'
+
+
+class SearchQueryField(Field):
+
+    def db_type(self, connection):
+        return 'tsquery'
+
+
+class SearchVector(Func):
+    function = 'to_tsvector'
+    arg_joiner = " || ' ' || "
+    output_field = SearchVectorField()
```

Comment:
Let me warn you about this. Internal methods (like `_resolve_output_field` and `_output_field_or_none`) don't care that you've overridden the regular output_field with your own class level version. It'll break in certain situations where it may be combined with another expression. I got bitten by this a few weeks ago when implementing Avg(duration) for Oracle.

That said, I'd love a solution that allowed this, because it's really really useful. Maybe `output_field_or_none` can check to see if the class has an attribute named output_field, and then skip any kind of output_field resolution.